changebot/github/github_api.py

```diff
@@ -99,6 +110,41 @@ def get_file_contents(self, path_to_file):
         contents_base64 = response.json()['content']
         return base64.b64decode(contents_base64).decode()
 
+    def get_user_config(self, path_to_file='astropybotrules.yaml'):
+        """
+        Load user configuration for bot.
+
+        Parameters
+        ----------
+        path_to_file : str
+            Configuration file in YAML format in the repository.
+            If not given or invalid, default is used.
+
+        Returns
+        -------
+        cfg : dict
+            Configuration parameters.
+
+        """
+        try:
+            file_content = self.get_file_contents(path_to_file)
+            cfg = yaml.load(file_content)
+        except Exception as e:
+            warnings.warn(str(e))
+            # Empty dict means calling code set the default
+            cfg = {}
+
+        return cfg
+
+    def get_config_from_cache(self, cfg_key, cfg_default):
+        """
+        Convenience method to extract user config from global cache.
+        """
+        cfg_cache_key = (self.repo, self.branch, self.installation)
+        # NOTE: Is there a chance that cache is reset at this point?
+        cfg = cfg_cache.get(cfg_cache_key, {})
+        return cfg.get(cfg_key, cfg_default)
```

Comment:
I think maybe a safer approach would be to not load the cache in ``__init__`` but instead do it here if the requested repo/branch/install are not in the cache.